aws-cfn/cross-service/resilient-workflow/README.md

```diff
@@ -0,0 +1,273 @@
+| --------- | ----------------------------------------- | ------------------------------------------------- |
+| `LB`      | The DNS Name of the primary Load Balancer | `curl` or Browser                                 |
+| `Key`     | The ID of a pem format private key in SSM | `ssh` after downloading from SSM                  |
+| `TGArn`   | The ARN of the target group of instances  | Check various additional information from the CLI |
+
+#### Demonstrate resiliency
+
+This part of the example demonstrates resiliency by simulating several kinds of failures.
+It uses Systems Manager parameters to update how the web server responds to requests. It
+also uses health checks to show how to make your web server more resilient to failure.
+
+Along with recommendations returned by the DynamoDB table, the web service includes the
+instance ID and Availability Zone so you can see how the load balancer distributes
+requests among the instances in the Auto Scaling group.
+
+The scenario takes the following steps. After editing the `params.json` file for each step, update the stack
+to see the changes by running an `update-stack` command.
+
+```
+aws cloudformation update-stack \
+  --stack-name resilience-demo \
+  --template-body file://./resilient-service.yaml \
+  --parameters file://./params.json \
+  --capabilities CAPABILITY_NAMED_IAM
+```
+
+1. **Initial state: healthy** — Sends requests to the endpoint to get recommendations and verify that instances
+   are healthy.
+2. **Broken dependency** — Sets a parameter that specifies a nonexistent DynamoDB table name. This simulates a
+   failure of the recommendation service. Requests for recommendations now return a failure
+   code. All instances still report as healthy because they only implement shallow health checks. For this
+   example, a shallow health check means the web server always reports itself as healthy as long as the
+   load balancer can connect to it.
+
+   Edit `params.json`. Add a new entry with `ParameterKey` as `SSMTableName` and `ParameterValue` as `bad_table`.
+   After updating, the service should report healthy but return `502` responses.
+
+3. **Static response** — Updates a parameter that prompts the web server to return a static response when the
+   recommendation service fails. Requests for recommendations now return a static response,
+   which is a better customer experience.
+
+   Edit `params.json`. Add a new entry with `ParameterKey` as `SSMFailure` and set `ParameterValue` to `static`.
+
+4. **Bad credentials** — Sets the table name parameter so the recommendations service succeeds, but also
+   updates one of the instances to use an instance profile that contains bad credentials.
+   Now, when the load balancer selects the bad instance to serve a request, it returns
+   a static response because it cannot access the recommendation service, but the other
+   instances return real recommendations.
```

Comment:
This doesn't seem to work without restarting the instance. From what I see, it hangs onto the credentials (probably because the boto3 instance is already created?) so removing the instance profile doesn't have any apparent effect.

Reply:
Done.